docs/dataflow_processing.md

```diff
@@ -0,0 +1,60 @@
+# Dataflow Processing
+
+NeonBee is to be considered a reactive dataflow engine. [Dataflows](https://en.wikipedia.org/wiki/Dataflow) strongly relate to stream processing or a whole programming paradigm called [dataflow programming](https://en.wikipedia.org/wiki/Dataflow_programming) and the [dataflow architecture](https://en.wikipedia.org/wiki/Dataflow_architecture). Dataflows can be thought of as a directed graph modeling the data flowing between operations and, in contrast to a classical von Neumann / control flow architecture, solely the availability of data will determine which action is executed next, not some iterative set of instructions. The reactive and asynchronous nature of Vert.x support this behavior very well, however resulting in the order of the graph nodes to be processed to be nondeterministic, depending on the data resolution strategy used.
+
+```
+      (Shopping Cart)
+       Entry Verticle
+       /           \
+      v             v
+(User Info) <- (Sales Items) -> (Pricing API)
+        \          /                  |
+         v        v             [web request]
+         (Database)
+             |
+         [mongo db]
+```
+
+Imagine in NeonBee every node or verticle (in the mathematical, not the Vert.x sense of the word) of the graph is represented by a `DataVerticle`. Every edge is determined by the verticles announced in the require phase, as described in the previous section. The contents of the node is determined in the second retrieval phase of data processing. The so called entry verticle, is a special name for the first `DataVerticle` invoked. It will finally, after all data was retrieved, determine the result of the operation. Imagine it as the root of the tree or graph. For one dataflow there is always exactly one entry verticle, depending whether or not an entry verticle should be exposed (e.g. via an endpoint, see later chapters), different notations can be used.
+
+## Data Resolution Strategies
+
+For retrieving data from the verticles, multiple so called resolution strategies can be used, however in one data stream, only one strategy is allowed. Mixing strategies is not possible, however in each phase of data processing, new requests can be sent. This might decrease the performance of the whole stream though. Imagine resolution strategies determining the order or process in which the graph of verticles is traversed. Each strategy comes with different traits and benefits, such as whether the strategy guarantees a deterministic order when certain verticles are executed, or wether different stages of verticles are called sequentially or not.
+
+### Recursive Strategy
+
+At the moment NeonBee offers only the recursive data resolution strategy. With the recursive strategy imagine each node and data requests to be processed in single stages:
+
+| Entry Stage | Stage 1 | Stage 2 | Stage 3 |
```

Comment:
I would prefer to add white spaces into the table, then it also looks good when it is not rendered.

Reply:
Tried it, and it is really cumbersome, because you have to manually space tables. And especially for very wide tables (like the ones I used below), where I have written like 100 characters, this becomes next to impossible... so I would recommend to stick to this "minimal" table design in markdown